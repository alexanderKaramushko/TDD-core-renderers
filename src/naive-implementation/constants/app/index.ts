export const appContainer = document.getElementById('app');

export const snakeSpeed = 5000;
