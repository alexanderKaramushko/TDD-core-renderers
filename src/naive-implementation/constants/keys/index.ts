export const enum Keys {
  'TOP' = 'w',
  'RIGHT' = 'd',
  'LEFT' = 'a',
  'BOTTOM' = 's',
}
