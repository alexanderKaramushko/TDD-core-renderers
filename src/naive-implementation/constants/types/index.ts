export const DITypes = {
  FieldFacade: Symbol.for('FieldFacade'),
  Food: Symbol.for('Food'),
  Segment: Symbol.for('Segment'),
  Snake: Symbol.for('Snake'),
};
