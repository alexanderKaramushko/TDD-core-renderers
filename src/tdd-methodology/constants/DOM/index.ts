export function getAppContainer(): Element | null {
  return document.getElementById('app');
}
