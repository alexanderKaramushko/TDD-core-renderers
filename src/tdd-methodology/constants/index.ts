export * from './DOM';
export * from './keys';
