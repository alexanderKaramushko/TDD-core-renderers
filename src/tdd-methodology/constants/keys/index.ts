export const enum Keys {
  'UP' = 'w',
  'RIGHT' = 'd',
  'LEFT' = 'a',
  'DOWN' = 's',
}
