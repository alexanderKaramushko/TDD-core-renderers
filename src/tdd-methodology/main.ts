import { Keys } from './constants';
import { GameInitializerModel } from './core/GameInitializer/types';

/**
 * @description entrypoint
 */
export default function main(gameInitializer: GameInitializerModel): void {
  gameInitializer.createGame();

  let direction: Keys = Keys.DOWN;

  setInterval(() => {
    if (direction === Keys.LEFT) {
      gameInitializer.moveLeft();
    }

    if (direction === Keys.UP) {
      gameInitializer.moveUp();
    }

    if (direction === Keys.RIGHT) {
      gameInitializer.moveRight();
    }

    if (direction === Keys.DOWN) {
      gameInitializer.moveDown();
    }
  }, 1000);

  window.addEventListener('keypress', (event: KeyboardEvent) => {
    direction = event.key as Keys;
  });
}
