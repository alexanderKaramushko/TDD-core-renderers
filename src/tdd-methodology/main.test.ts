import { getAppContainer } from './constants';
import main from './main';

/* eslint-disable no-undef */
describe('main', () => {
  it('should move segment', () => {
    document.body.innerHTML = '<div id="app"></div>';

    main();

    expect(getAppContainer()).toMatchSnapshot();
  });
});
