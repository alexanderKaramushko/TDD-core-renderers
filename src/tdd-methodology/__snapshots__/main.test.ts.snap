// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should move segment 1`] = `
<div
  id="app"
>
  <div
    id="field"
    style="left: 0px; top: 0px; width: 200px; height: 200px;"
  >
    <div
      class="segment"
      style="left: 0px; top: 10px;"
    />
    <div
      class="segment"
      style="left: 0px; top: 20px;"
    />
    <div
      class="segment"
      style="left: 0px; top: 30px;"
    />
    <div
      class="segment"
      style="left: 0px; top: 40px;"
    />
  </div>
</div>
`;
